File: cloudflow-backend/cloudflow-service-oa/src/main/java/com/cloudflow/oa/domain/Visitor.java
package com.cloudflow.oa.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.Version;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 访客预约实体类
 * 对应表：oa_visitor
 */
@Data
@TableName("oa_visitor")
public class Visitor implements Serializable {
    private static final long serialVersionUID = 1L;

    /** 访客记录ID */
    @TableId(value = "visitor_id", type = IdType.AUTO)
    private Long visitorId;

    /** 租户ID */
    private Long tenantId;

    /** 访客姓名 */
    private String visitorName;

    /** 访客电话 */
    private String visitorPhone;

    /** 访客单位 */
    private String visitorCompany;

    /** 来访人数 */
    private Integer visitorCount;

    /** 身份证号(脱敏存储) */
    private String idCard;

    /** 来访事由 */
    private String visitReason;

    /** 被访人ID */
    private Long hostId;

    /** 被访人姓名 */
    private String hostName;

    /** 被访人部门 */
    private String hostDept;

    /** 预约来访日期 */
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate visitDate;

    /** 预计到达时间 */
    private String visitTimeStart;

    /** 预计离开时间 */
    private String visitTimeEnd;

    /** 实际到达时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime actualArrive;

    /** 实际离开时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime actualLeave;

    /** 访问区域 */
    private String visitArea;

    /** 车牌号 */
    private String carPlate;

    /** 携带物品 */
    private String belongings;

    /** 访客照片URL */
    private String photoUrl;

    /** 通行证编号 */
    private String passCode;

    /** 工作流实例ID */
    private String processInstanceId;

    /** 状态(PENDING/APPROVING/APPROVAL_FAILED/REJECTED/CONFIRMED/ARRIVED/COMPLETED/CANCELLED) */
    private String status;

    /** 备注 */
    private String remark;

    /** 删除标志 */
    private Integer deleted;

    @Version
    private Integer version;
    /** 创建者 */
    private String createBy;

    /** 创建时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    /** 更新者 */
    private String updateBy;

    /** 更新时间 */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;
}

File: cloudflow-frontend/src/services/api/visitor.ts
import request from './request';
import { PageResult } from '@/types';

/** 访客预约接口类型 */
export interface Visitor {
  visitorId?: number;
  tenantId?: number;
  visitorName: string;
  visitorPhone?: string;
  visitorCompany?: string;
  visitorCount?: number;
  idCard?: string;
  visitReason: string;
  hostId: number;
  hostName?: string;
  hostDept?: string;
  visitDate: string;
  visitTimeStart?: string;
  visitTimeEnd?: string;
  actualArrive?: string;
  actualLeave?: string;
  visitArea?: string;
  carPlate?: string;
  belongings?: string;
  photoUrl?: string;
  passCode?: string;
  status?: string;
  remark?: string;
  createTime?: string;
}

/** 访客管理 API */
export const visitorApi = {
  list: (params: { pageNum?: number; pageSize?: number; status?: string; visitorName?: string; visitDate?: string }) =>
    request.get('/oa/visitor/list', { params }) as Promise<PageResult<Visitor>>,
  getInfo: (id: number) => request.get(`/oa/visitor/${id}`) as Promise<Visitor>,
  add: (data: Visitor) => request.post('/oa/visitor', data),
  edit: (data: Visitor) => request.put('/oa/visitor', data),
  remove: (ids: number[]) => request.delete(`/oa/visitor/${ids.join(',')}`),
  confirm: (id: number) => request.put(`/oa/visitor/confirm/${id}`),
  checkIn: (id: number) => request.put(`/oa/visitor/checkin/${id}`),
  checkOut: (id: number) => request.put(`/oa/visitor/checkout/${id}`),
  cancel: (id: number) => request.put(`/oa/visitor/cancel/${id}`),
  qrCode: (id: number) => request.get<Blob>(`/oa/visitor/${id}/qrcode`, {
    responseType: 'blob',
    silent: true,
  }),
};
